packages/devtools_app/lib/src/memory/flutter/memory_controller.dart

```diff
@@ -507,31 +680,207 @@ class MPChartData {
+  final List<Entry> other = <Entry>[];
+
+  /// Datapoint entries for system size value.
+  final List<Entry> system = <Entry>[];
+
+  /// Datapoint entries for total size value.
+  final List<Entry> total = <Entry>[];
+
+  /// Prune entries plotted based on number of sample
+  /// minutes to display.
+  void prune(int minutesToDisplay) {
+    assert(minutesToDisplay > 0);
+
+    final len = javaHeap.length;
+
+    // Entries should match.
+    // All entries lengths are the same.
+    assert(len == javaHeap.length &&
+        len == nativeHeap.length &&
+        len == code.length &&
+        len == stack.length &&
+        len == graphics.length &&
+        len == other.length &&
+        len == system.length &&
+        len == total.length);
+    // All entries starting timestamps match,
+    assert(javaHeap.isNotEmpty &&
+        javaHeap[0].x == nativeHeap[0].x &&
+        nativeHeap[0].x == code[0].x &&
+        code[0].x == stack[0].x &&
+        stack[0].x == graphics[0].x &&
+        graphics[0].x == other[0].x &&
+        other[0].x == system[0].x &&
+        system[0].x == total[0].x);
+    // All entries ending timestamps match.
+    assert(javaHeap.isNotEmpty &&
+        javaHeap[len - 1].x == nativeHeap[len - 1].x &&
+        nativeHeap[len - 1].x == code[len - 1].x &&
+        code[len - 1].x == stack[len - 1].x &&
+        stack[len - 1].x == graphics[len - 1].x &&
+        graphics[len - 1].x == other[len - 1].x &&
+        other[len - 1].x == system[len - 1].x &&
+        system[len - 1].x == total[len - 1].x);
+
+    // Compute a new starting index from length - N minutes.
+    final timeLastSample = javaHeap.last.x;
+    var index = len - 1;
+    for (; index > 0; index--) {
+      final sample = javaHeap[index];
+      final timestamp = sample.x;
+
+      if ((timeLastSample - timestamp) > minutesToDisplay) break;
+    }
+
+    if (index != 0) {
+      javaHeap.removeRange(0, index);
+      nativeHeap.removeRange(0, index);
+      code.removeRange(0, index);
+      stack.removeRange(0, index);
+      graphics.removeRange(0, index);
+      other.removeRange(0, index);
+      system.removeRange(0, index);
+      total.removeRange(0, index);
+    }
+  }
+
+  /// Add each entry to its corresponding trace.
+  void addTraceEntries({
+    Entry javaValue,
+    Entry nativeValue,
+    Entry codeValue,
+    Entry stackValue,
+    Entry graphicsValue,
+    Entry otherValue,
+    Entry systemValue,
+    Entry totalValue,
+    int minutesToDisplay,
+  }) {
+    if (!_pruning &&
+        javaHeap.isNotEmpty &&
+        (javaValue.x - javaHeap.first.x) > minutesToDisplay) {
+      _pruning = true;
+
+//      assert(javaValue.x - javaHeap.last.x <= minutesToDisplay);
+    }
+
+    if (_pruning) {
+      javaHeap.removeAt(0);
+      nativeHeap.removeAt(0);
+      code.removeAt(0);
+      stack.removeAt(0);
+      graphics.removeAt(0);
+      other.removeAt(0);
+      system.removeAt(0);
+      total.removeAt(0);
```

Comment:
can you keep a list of entries so that when you have to do blanket operations like this, you can instead write
```
for (var entry in entries) {
  entry.removeAt(0);
}
```
or below
```
for (var entry in entries) {
  entry.clear();
}
```

Reply:
The order of these traces are important I'm not sure what to be gained by losing the order here.  I'll think about how to fix this but right now I most comfortable all seeing the traces (there are lots of them).  Especially, as I make more changes and look at exposing display interval, etc to all users of charts.  I'll add a TODO for now to look at eliminating the explicit remove, clear, add, etc.